File: Insertion sort/python/insertionSort.py
class InsertionSort:
    def __init__(self, arr: list = None) -> None:
        self.arr = arr

    def sort(self, arr: list = None) -> None:
        if (arr != None):
            self.arr = arr

        for i in range(1, len(self.arr)):
            key = self.arr[i]
            j = i-1
            while (j >= 0 and self.arr[j] > key):
                self.arr[j+1] = self.arr[j]
                j -= 1

            self.arr[j+1] = key

    def display(self) -> None:
        if self.arr == None:
            return
        print("[", end="")
        for index, i in enumerate(self.arr):
            print(i, end="" if (index == len(self.arr)-1) else ", ")
        print("]")

File: Insertion sort/java/src/InsertionSort.java
public class InsertionSort {

    int[] arr;

    InsertionSort() {
    }

    InsertionSort(int[] arr) {
        this.arr = arr;
    }

    void sort() {
        int key, j;
        for (int i = 1; i < arr.length; i++) {
            key = arr[i];
            j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                --j;
            }
            arr[j + 1] = key;
        }
    }

    void sort(int[] arr) {
        this.arr = arr;
        sort();
    }

    void display() {
        if (arr != null) {

            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                System.out.print(arr[i] + ((i < arr.length - 1) ? ", " : ""));
            }
            System.out.println("]");
        }
    }
}
